packages/devtools/lib/src/timeline/timeline.dart

```diff
@@ -236,6 +258,63 @@ class TimelineScreen extends Screen {
     _updateListeningState();
   }
 
+  void loadFromImport(Map<String, dynamic> json) {
+    pauseButton.attribute('hidden', true);
+    resumeButton.attribute('hidden', true);
+    clearButton.attribute('hidden', true);
+    exportButton.attribute('hidden', true);
+    exitImportModeButton.attribute('hidden', false);
+
+    final List<dynamic> events = json['traceEvents'] ?? [];
+    final traceEvents = events
+        .cast<Map<String, dynamic>>()
+        .map((trace) => TraceEvent(trace))
+        .toList();
+    final Map<String, dynamic> cpuProfile = json['cpuProfile'] ?? {};
+
+    if (traceEvents.isEmpty && cpuProfile.isEmpty) {
+      framework.toast('Imported file does not contain timeline data.');
+      _exitImportMode();
+      return;
+    }
+
+    _clearTimeline();
+
+    if (traceEvents.isNotEmpty) {
+      timelineController.loadTimelineFromImport(traceEvents, cpuProfile);
+    } else {
+      framesBarChart.attribute('hidden', true);
+    }
+
+    if (cpuProfile.isNotEmpty) {
+      // TODO(kenzie): load CPU profile.
+    } else {
+      _destroySplitter();
+    }
+  }
+
+  void _exitImportMode() {
+    pauseButton.attribute('hidden', false);
+    resumeButton.attribute('hidden', false);
+    clearButton.attribute('hidden', false);
+    exportButton.attribute('hidden', false);
+    exitImportModeButton.attribute('hidden', true);
+
+    _clearTimeline();
+    _destroySplitter();
+
+    timelineController.exitImportMode();
+
+    if (serviceManager.connectedApp == null) {
```

Comment:
this seems like logic that should be in the core framework instead of here. I expect this applies any time we are exiting offline mode not just for the timeline.